### ISSUE TITLE
Not able to do...much

### ISSUE THREAD

**Issue by Kirk-TECHeGO** [opendns/autotask-php#44]
I'm struggling to get this package to do anything at all. Tried the basic tutorial and I'm getting a bunch of null values in the result. Is there some options inside of AutoTask I have to enable for the client first? I'm desperate for some advice please, here's what I've tried:

```
 <?php
   $username = "username@domain"; 
    $password = "password";
    $authWsdl = 'https://webservices.autotask.net/atservices/1.5/atws.wsdl";
    $opts = array("trace" => 1);
    $client = new ATWS\Client($authWsdl, $opts);
    $zoneInfo = $client->getZoneInfo($username);


    $authOpts = array(
        'login' => $username,
        'password' => $password,
        'trace' => 1,   // Allows us to debug by getting the XML requests sent
    );
    $wsdl = str_replace('.asmx', '.wsdl', $zoneInfo->getZoneInfoResult->URL);
    $client = new ATWS\Client($wsdl, $authOpts);

// Instantiate a Query object, designed to make complex
// queries simple.
    $query = new ATWS\AutotaskObjects\Query('Contact');

    $firstnameField = new ATWS\AutotaskObjects\QueryField('firstname');
    $firstnameField->addExpression('Equals', 'John');

    $query->addField($firstnameField);

// If you want to debug the XML produced by the Query object
   $result.=var_dump($query->asXml());
   $result.= " | ";
// Print the results of the query

    $result.=var_dump($client->query($query));
    $result.= " | ";

    $result.="authWsdl: $authWsdl |";
    $result.="client: $client |";
    $result.="zoneInfo: $zoneInfo |";
```

> "result": " |  | authWsdl: https://webservices.autotask.net/atservices/1.5/atws.wsdl |client:  |zoneInfo:  |"

**Comment by reynolek** [comment]
`var_dump` does not return the string it would output, [it is a void function](http://php.net/manual/en/function.var-dump.php).

I copy/pasted in your code there, removed all of your `$result` code to build some giant string, and used my credentials and it worked fine. See the code below. The end result is basically the exact example. Was there any reason you decided to tried to concat everything into a string?

```php
<?php
require_once '/opt/website/config/cli_bootstrap.php';

$username = "username@domain";
$password = "password";
$authWsdl = "https://webservices.autotask.net/atservices/1.5/atws.wsdl";
$opts = array("trace" => 1);
$client = new ATWS\Client($authWsdl, $opts);
$zoneInfo = $client->getZoneInfo($username);


$authOpts = array(
    'login' => $username,
    'password' => $password,
    'trace' => 1,   // Allows us to debug by getting the XML requests sent
);
$wsdl = str_replace('.asmx', '.wsdl', $zoneInfo->getZoneInfoResult->URL);
$client = new ATWS\Client($wsdl, $authOpts);

// Instantiate a Query object, designed to make complex
// queries simple.
$query = new ATWS\AutotaskObjects\Query('Contact');

$firstnameField = new ATWS\AutotaskObjects\QueryField('firstname');
$firstnameField->addExpression('Equals', 'John');

$query->addField($firstnameField);

// Print the results of the query
$result = $client->query($query);

print_r($result);
```

**Comment by Kirk-TECHeGO** [comment]
Thanks! 

Two reasons: I only partially know what I'm doing and also partially because of the environment I'm running the scripts in: dreamfactory. Sorry for my var_dump ignorance, and thank you for your support. Totally works now. I'll buy you a beer or something if I ever get the chance.

**Comment by reynolek** [comment]
Best of luck with the library -- Feel free to open another issue if you run into a road block.

**Comment by Kirk-TECHeGO** [comment]
In this example you used the entity called Contact. I see inside Contact.php there is a variable called $FirstName. After that we created a QueryField with the syntax 'firstname'. 

How are we to know the correct field name to use in a QueryField please? For example if I did a Query('Ticket') should I then assume QueryField('ticketnumber') is the correct syntax related to the $TicketNumber?

**Comment by reynolek** [comment]
If I remember correctly, the string passed to QueryField is case insensitive. So `'ticketnumber'` would work fine.

However if you wanted to set the `TicketNumber` class variable, you would need to do `$contact->TicketNumber = 12345`

**Comment by Kirk-TECHeGO** [comment]
I'm loving this library now! Is there an example anywhere of how to do an update to entity you found in your query?

**Comment by Kirk-TECHeGO** [comment]
So I'm trying to update a ticket. I've taken this approach.

First I look for it and store all the found values in new ticket:

```php
    $queryResult = $client->query($query);
    $updatedTicket = new ATWS\AutotaskObjects\Ticket();
    $updatedTicket->id = $queryResult->queryResult->EntityResults->Entity->id;
    $updatedTicket->Title = $queryResult->queryResult->EntityResults->Entity->Title;

//add rest of the fields from the query
...
```
Then I set the values I want to update:
```php
    $updateTicket->Priority = $priority;

```
I've then try to update it a couple different ways:
```php
$client->update($updateTicket);

```
or

```php
    $updateField = new ATWS\AutotaskObjects\UpdateParam($updatedTicket);
    $result = $client->update($updatedTicket);
```
of something like
```php
    $updateField = new ATWS\AutotaskObjects\UpdateParam('Priority');
    $result = $client->update($updatedTicket);
```
or half a dozen other variants.

Could you help me understand what I'm not understanding please?

**Comment by reynolek** [comment]
I dont have any examples immediately available -- can you read through this issue #1 and see if it helps get you going?
